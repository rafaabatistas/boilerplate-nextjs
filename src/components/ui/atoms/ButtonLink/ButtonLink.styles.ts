import styled from 'styled-components';

export const Link = styled.a`
  color: white;
  font-size: 1.8rem;
  text-decoration: none;
  transition: all 0.5s ease-out;

  &:hover {
    color: #adadad;
  }
`;
