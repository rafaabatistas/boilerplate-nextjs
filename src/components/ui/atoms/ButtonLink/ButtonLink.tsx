import * as S from './ButtonLink.styles';

type ButtonLinkProps = {
  link: string;
  children: React.ReactNode;
};

export const ButtonLink = ({ link, children }: ButtonLinkProps) => {
  return (
    <S.Link href={link} rel="noreferrer noopener" target="_blank">
      {children}
    </S.Link>
  );
};
