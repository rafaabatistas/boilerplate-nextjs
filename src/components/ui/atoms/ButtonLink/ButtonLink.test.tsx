import { render, screen } from '@testing-library/react';

import { ButtonLink } from './ButtonLink';

describe('<ButtonLink />', () => {
  it('Deve renderizar o componente <ButtonLink />', () => {
    const props = {
      link: 'https://github.com/rafaabatistas/boilerplate-nextjs',
      children: 'GitHub'
    };
    //Arrange
    const { container } = render(<ButtonLink {...props} />);
    //Assert
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Verifica se o componente <ButtonLink /> Ã© um link', () => {
    const props = {
      link: 'https://github.com/rafaabatistas/boilerplate-nextjs',
      children: 'GitHub'
    };
    render(<ButtonLink {...props} />);
    //Arrange
    const Link = screen.getByRole('link');
    //Assert
    expect(Link).toHaveAttribute('href');
  });
});
