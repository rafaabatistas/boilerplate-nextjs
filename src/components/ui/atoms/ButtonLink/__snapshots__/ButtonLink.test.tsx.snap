// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonLink /> Deve renderizar o componente <ButtonLink /> 1`] = `
.c0 {
  color: white;
  font-size: 1.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.5s ease-out;
  transition: all 0.5s ease-out;
}

.c0:hover {
  color: #adadad;
}

<a
  class="c0"
  href="https://github.com/rafaabatistas/boilerplate-nextjs"
  rel="noreferrer noopener"
  target="_blank"
>
  GitHub
</a>
`;
